lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+	attr_accessor :cells, :index
+
+	def initialize
+	    @cells = []
+		9.times{@cells << " "}
+	end
+
+	def reset!
+		@cells.clear
+		9.times{@cells << " "}
+	end	 
+
+	def display
+		puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+		puts "-----------"
+		puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+		puts "-----------"
+		puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+	end
+
+	def input_to_index(input)
+		@index = input.to_i - 1
+	end
+
+	def position(input)
+		@cells[input_to_index(input)]
+	end
+
+	def full?
+		!@cells.include?(" ")
+	end
+
+	def turn_count
+		@cells.collect{|i| i if i != " " }.compact.size + 1
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```